sdrpp: 1.1.0 -> 1.2.1

importantblimp opened the pull request:
Fixes #416080 

<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [Nixpkgs 25.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/doc/release-notes/rl-2511.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/doc/manual/release-notes/rl-2505.section.md) Nixpkgs Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
- [NixOS 25.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2511.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) NixOS Release notes)
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md), [pkgs/README.md](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md), [maintainers/README.md](https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md) and other contributing documentation in corresponding paths.

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

importantblimp commented:
I had a look through the commits since the last update and couldn't see anything substantial.

The program runs:
![image](https://github.com/user-attachments/assets/f2ecdaf9-ec9f-4168-9398-b4bf84001bf2)

Though I don't have hardware to test it. @sikmir or @dgramop I'd appreciate if you could help with that.

sikmir commented:
Both 1.1.0 and 1.2.1 failed to run on macOS with:

sikmir commented on pkgs/by-name/sd/sdrpp/package.nix:
```diff
@@ -69,9 +69,10 @@ stdenv.mkDerivation rec {
-  git_date = "2024-01-22";
-  version = "1.1.0-unstable-" + git_date;
+  git_hash = "a94e2d6712a6e5c80d6f6f27052e38be8a2ae0a9";
+  git_date = "2024-06-25";
```
```suggestion
  git_date = "2025-06-25";
```

sikmir commented:
Looks like `hardeningDisable = lib.optional stdenv.cc.isClang "format";` is not needed any more. Build is OK without it on darwin.

sikmir commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 420930`
Commit: `de79c8b50cacc4931bdfd76ed9c021d613d9a6c4`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>sdrpp</li>
  </ul>
</details>

importantblimp commented:
> Both 1.1.0 and 1.2.1 failed to run on macOS with:
> [ERROR] Glfw Error 65548: Cocoa: Regular windows do not have icons on macOS

What's the process here? Set `broken = stdenv.hostPlatform.isDarwin;`?

dgramop commented:
I need to test and debug again on an ARM Mac, which I can do next week. Hopefully broken will be temporary & quickly replaced.

rafaelrc7 commented:
@sikmir I found two issues in the upstream repo talking about crashes on MacOS: https://github.com/AlexandreRouma/SDRPlusPlus/issues/1558 https://github.com/AlexandreRouma/SDRPlusPlus/issues/1600 . Both were closed with the dev claiming the issues came from dependencies and not sdrpp itself. I don't know if it is still an issue (I don't have a mac).

importantblimp committed:
sdrpp: 1.1.0 -> 1.2.1

The ARM MacOS build has a few bugs (non-exhaustive) for version 1.2.1:
https://github.com/AlexandreRouma/SDRPlusPlus/issues/1667
https://github.com/AlexandreRouma/SDRPlusPlus/issues/1638
https://github.com/AlexandreRouma/SDRPlusPlus/issues/1431

sikmir commented:
> @sikmir I found two issues in the upstream repo talking about crashes on MacOS: [AlexandreRouma/SDRPlusPlus#1558](https://github.com/AlexandreRouma/SDRPlusPlus/issues/1558) [AlexandreRouma/SDRPlusPlus#1600](https://github.com/AlexandreRouma/SDRPlusPlus/issues/1600) . Both were closed with the dev claiming the issues came from dependencies and not sdrpp itself. I don't know if it is still an issue (I don't have a mac).
> 
> Both show the same `Cocoa: Regular windows do not have icons on macOS` error in the logs, but I also don't know if this is even the relevant error.

I have no idea how to fix it, so I'm fine with marking as broken on darwin.

Noderyos commented on pkgs/by-name/sd/sdrpp/package.nix:
```diff
@@ -90,7 +91,7 @@ stdenv.mkDerivation rec {
     # Since the __TIME_ and __DATE__ is canonicalized in the build,
     # use our qualified version shown in the programs window title.
-    substituteInPlace core/src/version.h --replace "1.1.0" "$version"
+    substituteInPlace core/src/version.h --replace "1.1.0" "$version" # Upstream haven't updated their version number from 1.1.0
```
Hi, nix relies on hash to determine if the source code needs to be downloaded, if the source code is already present in nix store, then the derivation commit is changed but not the hash, it keeps the old version of the source code.

rafaelrc7 replied:
While you update the git rev, you forgot to update the hash, that still is

```
hash = "sha256-R4xWeqdHEAaje37VQaGlg+L2iYIOH4tXMHvZkZq4SDU=";
```

importantblimp replied:
That's quite surprising, but good to know - thanks both

importantblimp commented:
> I have no idea how to fix it, so I'm fine with marking as broken on darwin.
It's unclear if it's broken on x86 too, but all of the reports I could see were aarch64 Macs so I've limited it to that for now.

I've also updated to the latest master. Could you please re-test?
I don't have the hardware for it, but the GUI functions

rafaelrc7 commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 420930`
Commit: `8945102f1322d205ff269f6d5cf5e3457d8b5c2e`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>sdrpp</li>
  </ul>
</details>
